lib/site_settings/validations.rb

```diff
@@ -53,4 +53,13 @@ def validate_enable_s3_uploads(new_val)
     validate_error :s3_upload_bucket_is_required if new_val == "t" && SiteSetting.s3_upload_bucket.blank?
   end
 
+  def validate_backup_location(new_val)
+    return unless new_val == BackupLocationSiteSetting::S3
+    validate_error(:s3_backup_requires_s3_settings, setting_name: "s3_backup_bucket") if SiteSetting.s3_backup_bucket.blank?
+
+    unless SiteSetting.s3_use_iam_profile
+      validate_error(:s3_backup_requires_s3_settings, setting_name: "s3_access_key_id") if SiteSetting.s3_access_key_id.blank?
+      validate_error(:s3_backup_requires_s3_settings, setting_name: "s3_secret_access_key") if SiteSetting.s3_secret_access_key.blank?
```

Comment:
Just curious, are we testing this validation anywhere in the test suite?

Reply:
Nope. As far as I can tell we don't have any specs for that file.